pkg/planner/core/rule_partition_processor.go

```diff
@@ -852,39 +852,21 @@ func (*partitionProcessor) name() string {
 
 type lessThanDataInt struct {
 	data     []int64
+	unsigned bool
 	maxvalue bool
 }
 
 func (lt *lessThanDataInt) length() int {
 	return len(lt.data)
 }
 
-func compareUnsigned(v1, v2 int64) int {
-	switch {
-	case uint64(v1) > uint64(v2):
-		return 1
-	case uint64(v1) == uint64(v2):
-		return 0
-	}
-	return -1
-}
-
 func (lt *lessThanDataInt) compare(ith int, v int64, unsigned bool) int {
-	if ith == len(lt.data)-1 {
-		if lt.maxvalue {
-			return 1
-		}
-	}
-	if unsigned {
-		return compareUnsigned(lt.data[ith], v)
-	}
-	switch {
-	case lt.data[ith] > v:
+	// TODO: get an extra partition when `v` bigger than `lt.maxvalue``, but the result still correct.
```

Comment:
Can you elaborate on this comment?
Does not this mean that it there is a `MAXVALUE` as the last partition, it will always be bigger / fit the value?

Reply:
```
mysql> explain select * from t where a > 9223372036854775809;
+-------------------------+----------+-----------+-----------------------+-----------------------------------+
| id                      | estRows  | task      | access object         | operator info                     |
+-------------------------+----------+-----------+-----------------------+-----------------------------------+
| TableReader_9           | 0.00     | root      |                       | data:Selection_8                  |
| └─Selection_8           | 0.00     | cop[tikv] |                       | gt(test.t.a, 9223372036854775809) |
|   └─TableFullScan_7     | 10000.00 | cop[tikv] | table:t, partition:p1 | keep order:false, stats:pseudo    |
+-------------------------+----------+-----------+-----------------------+-----------------------------------+
3 rows in set, 1 warning (0.00 sec)

mysql> show create table t;
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                   |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `a` bigint(20) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
PARTITION BY RANGE (`a`)
(PARTITION `p0` VALUES LESS THAN (5086706),
 PARTITION `p1` VALUES LESS THAN (MAXVALUE)) |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

`9223372036854775809` already bigger than max value of signed bigint. In MySQL the plan is like 

```
mysql> explain select * from t where a > 9223372036854775809;
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+------------------+
| id | select_type | table | partitions | type | possible_keys | key  | key_len | ref  | rows | filtered | Extra            |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+------------------+
|  1 | SIMPLE      | NULL  | NULL       | NULL | NULL          | NULL | NULL    | NULL | NULL |     NULL | Impossible WHERE |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+------------------+
1 row in set, 1 warning (0.00 sec)
```

Reply:
Ah, you mean that it should return table dual (or no set of partitions), not an extra partition?

Reply:
Yes.